src/main/java/application/MenuController.java

```diff
@@ -156,10 +141,9 @@ public EntityDetailListResponse getDetails(@RequestBody SessionNavigationBean se
             throw new RuntimeException("Could not find case with ID " + detailSelection);
         }
 
-        return new EntityDetailListResponse(
-                entityScreen,
-                menuSession.getSessionWrapper().getEvaluationContext(),
-                reference
+        return setLocationNeeds(
```

Comment:
What I was trying to say here was that it would be nice to move these `setLocationNeeds` calls from being called at the end of [getDetails](https://github.com/dimagi/formplayer/blob/0a3d04e32cad9cf195c828673235764c091cced9/src/main/java/services/MenuSessionRunnerService.java#L318-L318) and [advanceSessionWIthSelections](https://github.com/dimagi/formplayer/blob/0a3d04e32cad9cf195c828673235764c091cced9/src/main/java/services/MenuSessionRunnerService.java#L147-L147) rather than after these methods return. However, I'm now noticing that the call paths for these four usages are all slightly different in a way that would make this not clean... so maybe worth worrying about in a future refactor of this (messy) code